pkg/writer/junit_xml.go

```diff
@@ -0,0 +1,192 @@
+	testSuiteName  string          = "TERRASCAN_POLICY_SUITE"
+	testSuitesName string          = "TERRASCAN_POLICY_SUITES"
+)
+
+// JUnitTestSuites is a collection of JUnit test suites.
+type JUnitTestSuites struct {
+	XMLName  xml.Name `xml:"testsuites"`
+	Tests    int      `xml:"tests,attr"`
+	Name     string   `xml:"name,attr"`
+	Failures int      `xml:"failures,attr"`
+	Time     string   `xml:"time,attr"`
+	Suites   []JUnitTestSuite
+}
+
+// JUnitTestSuite is a single JUnit test suite which may contain many testcases.
+type JUnitTestSuite struct {
+	XMLName    xml.Name        `xml:"testsuite"`
+	Tests      int             `xml:"tests,attr"`
+	Failures   int             `xml:"failures,attr"`
+	Time       string          `xml:"time,attr"`
+	Name       string          `xml:"name,attr"`
+	Package    string          `xml:"package,attr"`
+	Properties []JUnitProperty `xml:"properties>property,omitempty"`
+	TestCases  []JUnitTestCase
+}
+
+// JUnitTestCase is a single test case with its result.
+type JUnitTestCase struct {
+	XMLName   xml.Name `xml:"testcase"`
+	Classname string   `xml:"classname,attr"`
+	Name      string   `xml:"name,attr"`
+	Severity  string   `xml:"severity,attr"`
+	Category  string   `xml:"category,attr"`
+	// omit empty time because today we do not have this data
+	Time        string            `xml:"time,attr,omitempty"`
+	SkipMessage *JUnitSkipMessage `xml:"skipped,omitempty"`
+	Failure     *JUnitFailure     `xml:"failure,omitempty"`
+}
+
+// JUnitSkipMessage contains the reason why a testcase was skipped.
+type JUnitSkipMessage struct {
+	Message string `xml:"message,attr"`
+}
+
+// JUnitProperty represents a key/value pair used to define properties.
+type JUnitProperty struct {
+	Name  string `xml:"name,attr"`
+	Value string `xml:"value,attr"`
+}
+
+// JUnitFailure contains data related to a failed test.
+type JUnitFailure struct {
+	Message  string `xml:"message,attr"`
+	Type     string `xml:"type,attr"`
+	Contents string `xml:",chardata"`
+}
+
+func newJunitTestSuites(summary results.ScanSummary) JUnitTestSuites {
+	return JUnitTestSuites{
+		Tests:    summary.TotalPolicies,
+		Name:     testSuitesName,
+		Failures: summary.ViolatedPolicies,
+		Time:     fmt.Sprint(summary.TotalTime),
+	}
+}
+
+func newJunitTestSuite(summary results.ScanSummary) JUnitTestSuite {
+	return JUnitTestSuite{
+		Name:     testSuiteName,
+		Tests:    summary.TotalPolicies,
+		Time:     fmt.Sprint(summary.TotalTime),
+		Failures: summary.ViolatedPolicies,
+		Package:  summary.ResourcePath,
+		Properties: []JUnitProperty{
+			{
+				Name:  "Terrascan Version",
+				Value: version.Get(),
+			},
+		}}
+}
+
+func init() {
+	RegisterWriter(junitXMLFormat, JUnitXMLWriter)
+}
+
+// JUnitXMLWriter writes scan summary in junit xml format
+func JUnitXMLWriter(data interface{}, writer io.Writer) error {
+	output, ok := data.(policy.EngineOutput)
+	if !ok {
+		return fmt.Errorf("incorrect input for JunitXML writer, supportted type is policy.EngineOutput")
```

Comment:
typo => supportted. 

Also, the error message seems good for a debug level log but not for an error level log as the user should not be bothered to understand what is `policy.EngineOutput`.

Reply:
The error is only for the developers, so that developers would know what type to pass

Reply:
alrighty

Reply:
can you correct the type though? @patilpankaj212